src/commonTest/kotlin/ToolSerializationTest.kt

```diff
@@ -0,0 +1,57 @@
+package io.modelcontextprotocol.kotlin.sdk
+
+import io.kotest.assertions.json.shouldEqualJson
+import io.modelcontextprotocol.kotlin.sdk.shared.McpJson
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.buildJsonObject
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class ToolSerializationTest {
+
+    // see https://docs.anthropic.com/en/docs/build-with-claude/tool-use
+    /* language=json */
+    private val getWeatherToolJson = """
```

Comment:
We could use `buildJsonObject` here: makhttps://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/build-json-object.html